### ISSUE TITLE
IDE0072 does not get properly reported (e.g. no `(TEnum)0` when `Nullable<TEnum>` gets switched and `null` case is present)

### ISSUE THREAD

**Issue by LukasGelke** [dotnet/roslyn#57523]
**Version Used**: 
- Visual Studio Professional 2019; Version 16.11.5
- .NET SDK 3.1.414

**Steps to Reproduce**:
I could not reproduce any of the following with `dotnet build`. It never shows any warning/error.

Open the following project/solution in Visual Studio (I even tried a new, clean rootsuffix). The switch expression has no error.
However, when commenting out the line `null => null,`, suddenly the analyzer kicks in.

**Program1.cs**
```cs
namespace Issue
{
  public class Program
  {
    public static int Main() => GetCuts(Shape.Circle) is null ? -1 : 0;

    static int? GetCuts(Shape? shape) => shape switch
    {
      null => null,
      Shape.Triangle => 3,
      Shape.Square or Shape.Rectangle => 4,
      _ => throw new System.NotSupportedException(),
    };
  }

  public enum Shape
  {
    Circle,
    Square,
    Rectangle,
    Triangle,
  }
}
```

**ConsoleApp1.csproj**
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <Nullable>enable</Nullable>
    <LangVersion>preview</LangVersion>
  </PropertyGroup>
</Project>
```

**.editorconfig**
```editorconfig
[*.cs]
dotnet_diagnostic.IDE0072.severity = error
```

**Expected Behavior**:
IDE0072 is raised (because of `Shape.Circle`)

**Actual Behavior**:
no error at all

**Things to note**:
- trying to reproduce this with top-level statements also had inconsistent results
  - It sometimes wouldn't even raise without `null`-case and as non-`Nullable<>`
  - even modifying whitespace (adding/removing lines in the top/middle/bottom) would randomly show/hide the error/warning
- `null` is not considered as a warnable/fixable case for `Nullable<TEnum>` - is this intentional?
- (when nullable) the codefix wants to add `Shape?.Circle` which is not valid C# syntax

**Comment by LukasGelke** [comment]
All of the same behavior applies to IDE0010 respectively:
```cs
static int? GetCuts(Shape? shape)
    {
      switch (shape)
      {
        case null: return null;
        case Shape.Triangle: return 3;
        case Shape.Square: case Shape.Rectangle: return 4;
        default: throw new System.NotSupportedException();
      };
    }
```

**Comment by ryanbrandenburg** [comment]
Design Meeting Notes:
> The compiler should tell us what "exhausted" means for a switch statement or expression via some API, and then we can use that to improve this analyzer.
